crates/rpcsrv/src/flight/handler.rs

```diff
@@ -140,11 +99,57 @@ impl FlightSessionHandler {
             .map_err(Status::from);
         Ok(Response::new(Box::pin(stream)))
     }
+
+    pub fn new(engine: &Arc<Engine>) -> Self {
+        Self {
+            engine: engine.clone(),
+            logical_plans: DashMap::new(),
+            sessions: DashMap::new(),
+        }
+    }
+
+    async fn get_or_create_ctx<T>(
+        &self,
+        request: &Request<T>,
+    ) -> Result<Arc<Mutex<Session>>, Status> {
+        let remote = request.remote_addr().unwrap();
+
+        let ip = remote.ip().to_string();
+        let port = remote.port().to_string();
+        let conn_key = ConnKey { ip, port };
+
+        if self.sessions.contains_key(&conn_key) {
+            let sess = self.sessions.get(&conn_key).unwrap().clone();
+            return Ok(sess);
+        }
+
+        let db_id = request
+            .metadata()
+            .get(DATABASE_HEADER)
+            .map(|s| Uuid::try_parse_ascii(s.as_bytes()).ok())
+            .flatten()
+            .unwrap_or_else(|| Uuid::new_v4());
+
+        let session_vars = SessionVars::default()
+            .with_database_id(db_id, datafusion::variable::VarType::System)
+            .with_force_catalog_refresh(true, datafusion::variable::VarType::System);
```

Comment:
is this a change over previous behaviors? 

I guess my question is really:

- how often do we create a new session in flight?
- does this mean we're forcing a catalog refresh more often now?

Reply:
should be the same number of sessions, just passing in the db id from cloud proxy now. The # of sessions is unchanged. 

> does this mean we're forcing a catalog refresh more often now?

it should be unchanged.